test/common/cro_cr_success.go

```diff
@@ -0,0 +1,128 @@
+package common
+
+import (
+	goctx "context"
+	"encoding/json"
+	"fmt"
+	crov1 "github.com/integr8ly/cloud-resource-operator/pkg/apis/integreatly/v1alpha1"
+	croTypes "github.com/integr8ly/cloud-resource-operator/pkg/apis/integreatly/v1alpha1/types"
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/resources/constants"
+	"k8s.io/apimachinery/pkg/types"
+	"testing"
+)
+
+const (
+	requestNameSpace  = "redhat-rhmi-operator"
+	requestUrl        = "/apis/integreatly.org/v1alpha1"
+	openShiftProvider = "openshift"
+	externalProvider  = "aws"
+)
+
+func TestCROPostGresSuccessfulState(t *testing.T, ctx *TestingContext) {
+	postGresToCheck := []string{
+		fmt.Sprintf("%s%s", constants.CodeReadyPostgresPrefix, InstallationName),
+		fmt.Sprintf("%s%s", constants.ThreeScalePostgresPrefix, InstallationName),
+		fmt.Sprintf("%s%s", constants.RHSSOPostgresPrefix, InstallationName),
+		fmt.Sprintf("%s%s", constants.RHSSOUserProstgresPrefix, InstallationName),
+		fmt.Sprintf("%s%s", constants.UPSPostgresPrefix, InstallationName),
+		constants.AMQAuthServicePostgres,
+	}
+
+	for _, postGresName := range postGresToCheck {
+
+		strategy := getResourceStrategy(t, ctx)
```

Comment:
I think the `getResourceStrategy` should be outside the loop, I think you already did this for the other tests

Reply:
if it's because of the AMQ Postgres than I would suggest to do something like this
```go
originalStrategy := getResourceStrategy(t, ctx)
for _, postGresName := range postGresToCheck {
    strategy := originalStrategy
```

Reply:
I was thinking something similar to this but I put it in the loop as I was thinking that for this to work, we always have to ensure the AMQ Postgres must be the last postgres to check as the strategy is only fetched once and would affect the other tests if this is not the case :thinking: 

I'm good with putting it outside the loop, if you think that is alright or isn't a concern ?

Reply:
I wouldn't give for granted that the AMQ is always the last, therefore my idea was to put the strategy in the `orignalStrategy` so that the original never gets overwritten. But putting the `getResourceStrategy` outside the loop is just a small improvement I would do because it would reduce the number of requests, but it doesn't change much else for the test so it's up to you

Reply:
Oh I see, yeah I mis-read that. Sounds good to me so :+1: